libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -476,6 +476,8 @@ void AP_Camera::take_picture()
  */
 void AP_Camera::update_trigger()
 {
+    update();
```

Comment:
It seems odd to me that update() is called from within update_trigger().  I think instead we should make update_trigger private and call it from within the public "update".